pkg/skaffold/build/dependencies.go

```diff
@@ -30,8 +30,56 @@ import (
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 )
 
-// DependenciesForArtifact returns the dependencies for a given artifact.
-func DependenciesForArtifact(ctx context.Context, a *latest.Artifact, cfg docker.Config, r docker.ArtifactResolver) ([]string, error) {
+// for testing
+var getDependenciesFunc = dependenciesForSingleArtifact
+
+type DependencyResolver interface {
+	DependenciesForArtifact(ctx context.Context, a *latest.Artifact) ([]string, error)
```

Comment:
The convention in our codebase seems to be to pass around pointers for all `SkaffoldConfig` objects so I followed the same. Otherwise there's no need for this to be a pointer.